pkg/generate/app/imagestreamlookup.go

```diff
@@ -143,6 +143,13 @@ func (r ImageStreamSearcher) Search(precise bool, terms ...string) (ComponentMat
 						}
 						foundOtherTags = true
 
+						// This check after foundOtherTags = true, because in the case that
+						// we have an ImageStream with all tags hidden, we should behave as
+						// if it didn't exist at all.
```

Comment:
ah i guess i see it's because if we didn't "find other tags" then it's going to create a match for the imagestream.

if we did find other tags (or lie about having found them), then it will not create that match.